talker/src/talker.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2019, ROSIN-project
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * This tutorial demonstrates simple sending of messages over the ROS system 
+ * using the rxros interface
+ */
+
+#include <rxros/rxros.h>
+#include "std_msgs/String.h"
+
+using namespace rxcpp::operators;
+using namespace rxros::operators;
+
+std_msgs::String mk_msg (const std::string& s)  // the missing factory
+{ 
+    std_msgs::String msg;
+    msg.data = std::string(s);
+    return msg;
+};
+
+
+int main(int argc, char **argv)
+{
+  rxros::init(argc, argv, "talker");
+  const std::string hello = "hello world ";
+
+  rxcpp::observable<>::interval (std::chrono::milliseconds (1000)) 
+    | map ([&](int i) 
+        { return mk_msg(hello + std::to_string(i)); })
+    | tap ([](const std_msgs::String& msg) 
+        { ROS_INFO ("%s", msg.data.c_str()); })
+    | publish_to_topic<std_msgs::String> ("/chatter", 1000);
```

Comment:
This may be my lack of experience with RxCPP showing, but does [this](https://github.com/rosin-project/rxros/blob/f170daeb34ca91d3e36f83c62510e415d9f1aacc/rxros/include/rxros/rxros.h#L300-L306) recreate the `publisher` every time `publish_to_topic(..)` is called?

If so, that would be a problem.

Reply:
It should not need to... but I will check how @henrik7264 has implemented this.  At this point no computation should be happening, only after the spin call.  This should just build a representation that is later run.  #tocheck

Reply:
@wasowski is your program working as you expect? 
I made some test this evening and the publisher should accordingly to my tests only be be created once. However, I found a bug in the publish_to_topic function. It is rather annoying, but the "observe_on" call will have to be replaced with a "subscribe_on" call in the publish_to_topic function.  I would actually not expect your talker to reach the rxros::spin() call - except the interval observable is running in a dedicated thread. 

PS: The second parameter to the from_topic and publish_to_topic is the stack size. 1000 is a rater large stack size - 10 is default.

Reply:
RE: behaviour, the example works well, but if there is a bug in RxRos we should create an issue there and handle it separately.  Have you @henrik7264  ? (I wouldn't like this to be lost)

RE: 1000 -- I realize it is idiotically high, but this is the value in the original ROS tutorial, so I don't want to change it.  I only want to change things that have to do with RxROS.